Dockerfile.osx

```diff
@@ -8,37 +8,20 @@ RUN if [ -z "${mono_version}" ]; then echo -e "\n\nargument mono-version is mand
     dnf clean all && \
     git clone https://github.com/tpoechtrager/osxcross.git && \
     cd /root/osxcross && \
+    git checkout 542acc2ef6c21aeb3f109c03748b1015a71fed63 && \
     ln -s /root/files/MacOSX10.14.sdk.tar.xz /root/osxcross/tarballs && \
-    UNATTENDED=1 ./build.sh ;\
-    cd /root && \
-    git clone https://github.com/tpoechtrager/apple-libtapi.git && \
-    cd apple-libtapi && \
-    INSTALLPREFIX=/root/osxcross/target ./build.sh && ./install.sh && \
-    cd /root && \
-    git clone https://github.com/tpoechtrager/cctools-port.git && \
-    cd cctools-port/cctools/ && \
-    ./configure --prefix=/root/osxcross/target --target=x86_64-apple-darwin17 --with-libtapi=/root/osxcross/target && \
-    make -j && make install && \
-    cd /root && \
-    ln -fs /root/osxcross/target/bin/x86_64-apple-darwin17-install_name_tool /root/osxcross/target/bin/install_name_tool && \
-    ln -fs /root/osxcross/target/bin/x86_64-apple-darwin17-ar /root/osxcross/target/bin/ar && \
-    cd /root/osxcross/ && \
-    ./build_compiler_rt.sh && \
-    export CLANG_LIB_DIR=$(clang -print-search-dirs | grep "libraries: =" | tr '=' ' ' | tr ':' ' ' | awk '{print $2}') && \
-    mkdir -p ${CLANG_LIB_DIR}/include && \
-    mkdir -p ${CLANG_LIB_DIR}/lib/darwin && \
-    cp -rv /root/osxcross/build/compiler-rt/include/sanitizer ${CLANG_LIB_DIR}/include && \
-    cp -v /root/osxcross/build/compiler-rt/build/lib/darwin/*.a ${CLANG_LIB_DIR}/lib/darwin && \
-    cp -v /root/osxcross/build/compiler-rt/build/lib/darwin/*.dylib ${CLANG_LIB_DIR}/lib/darwin
+    UNATTENDED=1 ./build.sh
+
+ENV OSXCROSS_ROOT=/root/osxcross
+ENV PATH="/root/osxcross/target/bin:${PATH}"
 
 RUN git clone https://github.com/mono/mono --branch mono-${mono_version} --single-branch && \
     cd mono && git submodule update --init && \
-    patch -p1 < /root/files/patches/fix-mono-configure.diff && \
-    export PATH=/root/osxcross/target/bin:$PATH && \
-    export CMAKE=/root/osxcross/target/bin/x86_64-apple-darwin17-cmake && \
-    ./autogen.sh --prefix=/root/dependencies/mono \
+    export CMAKE=/root/osxcross/target/bin/x86_64-apple-darwin18-cmake && \
+    NOCONFIGURE=1 ./autogen.sh && \
+    ./configure --prefix=/root/dependencies/mono \
         --build=x86_64-linux-gnu \
-        --host=x86_64-apple-darwin17 \
+        --host=x86_64-apple-darwin18 \
```

Comment:
Is it expected that we use darwin as host but linux as build? Shouldn't it be the other way around?

Reply:
It would also be great if we could eventually properly build the osx mono without having to mix it up with the Fedora installation from the `mono` dockerfile.

Reply:
According to `info autogen`:

‘--build=BUILD’
     The system on which the package is built.
‘--host=HOST’
     The system where built programs and libraries will run.

I built mono 6.4.0.198 for OSX on my local ubuntu system, before getting into dockers. I was using this: `--build=x86_64-linux-gnu --host=x86_64-apple-darwin19 `. I remember it complaining a lot and not building until I got it right.